File: 프로그래머스/소수찾기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

bool memo[1000000]{ false };

int solution(int n) {
	int answer = 0;
	for (int i = 2; i <= n; i++) {
		if (memo[i - 1])
			continue;
		answer++;
		for (int j = i + i; j <= n; j += i) {
			memo[j - 1] = true;
		}
	}
	return answer;
}

int main() {
	cout << solution(10);
}

File: 프로그래머스/소수찾기.py
def solution(n):
    answer = 0
    numset = set(range(2, n + 1))
    for i in range(2, n + 1):
        if i in numset:
            numset -= set(range(2 * i, n + 1, i))
    answer = len(numset)
    return answer
